gnark-prover/prover/circuit_utils.go

```diff
@@ -63,6 +78,25 @@ func (gadget InclusionProof) DefineGadget(api frontend.API) interface{} {
 	return currentHash
 }
 
+func (gadget NonInclusionProof) DefineGadget(api frontend.API) interface{} {
+	currentHash := make([]frontend.Variable, gadget.NumberOfUtxos)
+	for proofIndex := 0; proofIndex < gadget.NumberOfUtxos; proofIndex++ {
+		api.AssertIsDifferent(gadget.LeafLowerRangeValue[proofIndex], gadget.Value[proofIndex])
+		api.AssertIsLessOrEqual(gadget.LeafLowerRangeValue[proofIndex], gadget.Value[proofIndex])
```

Comment:
is there no AssertIsLess api?

Reply:
Standard api doesn't have AssertIsLess. We can use [cmp package](https://pkg.go.dev/github.com/consensys/gnark@v0.9.1/std/math/cmp), like:	
`api.AssertIsEqual(cmp.IsLess(api, gadget.LeafLowerRangeValue[proofIndex], gadget.Value[proofIndex]), 1)`
 but I can't say it's definitely cleaner.

Reply:
let's just keep it this way then